### ISSUE TITLE
ICE with let_else: trimmed_def_paths constructed

### ISSUE THREAD

**Issue by est31** [rust-lang/rust#91017]
### Code

I was working on #91018, a follow up for #89933, and was adding `let_else` to `rustc_mir_build` when I encountered the ICE.

The code is available in my local rustc fork on the [let_else_ice](https://github.com/est31/rust/tree/let_else_ice) branch. I've minimized it to the single invocation that causes the ICE.

### Error output

```
thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:53:62
```

<details><summary><strong>Full error log</strong></summary>
<p>

```
$ ./x.py check
Updating only changed submodules
Submodules updated in 0.01 seconds
    Finished dev [unoptimized] target(s) in 0.11s
warning: there have been changes to x.py since you last updated.
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, update `config.toml` to use `changelog-seen = 2` instead
Checking stage0 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.10s
Checking stage0 std test/bench/example targets (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.11s
Checking stage0 compiler artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Checking rustc_mir_build v0.0.0 (/home/user/src/rust/compiler/rustc_mir_build)
thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:53:62
thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:53:62
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


error: internal compiler error: unexpected panic
error: internal compiler error: unexpected panic


note: the compiler unexpectedly panicked. this is a bug.
note: the compiler unexpectedly panicked. this is a bug.


note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md
note: rustc 1.57.0-beta.2 (7e4c9eebd 2021-10-22) running on x86_64-unknown-linux-gnu


note: rustc 1.57.0-beta.2 (7e4c9eebd 2021-10-22) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z symbol-mangling-version=v0 -Z macro-backtrace -Z tls-model=initial-exec -Z unstable-options -Z binary-dep-depinfo -Z force-unstable-if-unmarked -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 -C linker=/usr/bin/clang -C link-args=-Wl,-rpath,$ORIGIN/../lib -C prefer-dynamic

note: compiler flags: -Z symbol-mangling-version=v0 -Z macro-backtrace -Z tls-model=initial-exec -Z unstable-options -Z binary-dep-depinfo -Z force-unstable-if-unmarked -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 -C linker=/usr/bin/clang -C link-args=-Wl,-rpath,$ORIGIN/../lib -C prefer-dynamic --crate-type lib
note: some of the compiler flags provided by cargo are hidden


note: some of the compiler flags provided by cargo are hidden
query stack during panic:

query stack during panic:
#0 [mir_borrowck] borrow-checking `build::matches::<impl at compiler/rustc_mir_build/src/build/matches/mod.rs:37:1: 816:2>::place_into_pattern`
#0 [mir_borrowck] borrow-checking `build::matches::<impl at compiler/rustc_mir_build/src/build/matches/mod.rs:37:1: 816:2>::place_into_pattern`
#1 [analysis] running analysis passes on this crate
#1 [analysis] running analysis passes on this crate
end of query stack
end of query stack
error: internal compiler error: trimmed_def_paths constructed
  |
  = note: delayed at    0: rustc_errors::Handler::delay_good_path_bug
             1: rustc_middle::ty::print::pretty::trimmed_def_paths
             2: rustc_data_structures::stack::ensure_sufficient_stack
             3: rustc_query_system::query::plumbing::try_execute_query
             4: rustc_query_system::query::plumbing::get_query
             5: <rustc_middle::ty::print::pretty::FmtPrinter<F> as rustc_middle::ty::print::Printer>::print_def_path
             6: rustc_middle::ty::print::Printer::default_print_def_path
             7: <rustc_middle::ty::print::pretty::FmtPrinter<F> as rustc_middle::ty::print::Printer>::print_def_path
             8: <rustc_middle::ty::print::pretty::FmtPrinter<F> as rustc_middle::ty::print::Printer>::print_type
             9: rustc_middle::ty::print::pretty::<impl core::fmt::Display for &rustc_middle::ty::TyS>::fmt
            10: core::fmt::write
                       at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/core/src/fmt/mod.rs:1163:17
            11: core::fmt::Write::write_fmt
                       at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/core/src/fmt/mod.rs:186:9
            12: alloc::fmt::format
                       at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/alloc/src/fmt.rs:579:5
            13: rustc_borrowck::diagnostics::BorrowedContentSource::describe_for_unnamed_place
            14: rustc_borrowck::diagnostics::move_errors::<impl rustc_borrowck::MirBorrowckCtxt>::report_move_errors
            15: rustc_borrowck::do_mir_borrowck
            16: rustc_infer::infer::InferCtxtBuilder::enter
            17: core::ops::function::FnOnce::call_once
            18: rustc_query_system::query::plumbing::try_execute_query
            19: <rustc_query_impl::Queries as rustc_middle::ty::query::QueryEngine>::mir_borrowck
            20: rustc_data_structures::sync::par_for_each_in
            21: rustc_session::utils::<impl rustc_session::session::Session>::time
            22: rustc_interface::passes::analysis
            23: rustc_query_system::query::plumbing::try_execute_query
            24: <rustc_query_impl::Queries as rustc_middle::ty::query::QueryEngine>::analysis
            25: rustc_interface::passes::QueryContext::enter
            26: rustc_interface::queries::<impl rustc_interface::interface::Compiler>::enter
            27: rustc_span::with_source_map
            28: scoped_tls::ScopedKey<T>::set
            29: std::sys_common::backtrace::__rust_begin_short_backtrace
            30: core::ops::function::FnOnce::call_once{{vtable.shim}}
            31: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
                       at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/alloc/src/boxed.rs:1691:9
            32: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
                       at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/alloc/src/boxed.rs:1691:9
            33: std::sys::unix::thread::Thread::new::thread_start
                       at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys/unix/thread.rs:106:17
            34: start_thread
                       at ./nptl/./nptl/pthread_create.c:435:8
            35: __clone
                       at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:100
          

thread 'rustc' panicked at 'no warnings or errors encountered even though `delayed_good_path_bugs` issued', compiler/rustc_errors/src/lib.rs:1165:13
stack backtrace:
   0:     0x7fe5f46dbafc - std::backtrace_rs::backtrace::libunwind::trace::h25fd93f23613b0bd
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x7fe5f46dbafc - std::backtrace_rs::backtrace::trace_unsynchronized::h5c84b72802e6f97b
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7fe5f46dbafc - std::sys_common::backtrace::_print_fmt::h2963c3d3cac93165
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7fe5f46dbafc - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hc11090b821447af0
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x7fe5f4738d7c - core::fmt::write::h4177df205e3207df
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/core/src/fmt/mod.rs:1163:17
   5:     0x7fe5f46cc065 - std::io::Write::write_fmt::h43248dd5a934053b
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/io/mod.rs:1696:15
   6:     0x7fe5f46ded50 - std::sys_common::backtrace::_print::he9298a8f9ec8c87d
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x7fe5f46ded50 - std::sys_common::backtrace::print::h21f708b17f3753fb
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x7fe5f46ded50 - std::panicking::default_hook::{{closure}}::hbf61d78ce94d8717
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:210:50
   9:     0x7fe5f46de8fb - std::panicking::default_hook::h17de5f695e938d3c
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:227:9
  10:     0x7fe5f4ebf5c1 - rustc_driver::DEFAULT_HOOK::{{closure}}::{{closure}}::h136d34559455780f
  11:     0x7fe5e8bd9819 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h85db76da50bfa969
  12:     0x7fe5e8bda76b - proc_macro::bridge::client::<impl proc_macro::bridge::Bridge>::enter::{{closure}}::{{closure}}::h1b552d664e4eed54
  13:     0x7fe5f46df569 - std::panicking::rust_panic_with_hook::hab1577e24779214f
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:628:17
  14:     0x7fe5f46df020 - std::panicking::begin_panic_handler::{{closure}}::h8c53778a9a259922
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:521:13
  15:     0x7fe5f46dbfa4 - std::sys_common::backtrace::__rust_end_short_backtrace::h24cd09d06b6955fd
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:139:18
  16:     0x7fe5f46def89 - rust_begin_unwind
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:517:5
  17:     0x7fe5f46a4261 - core::panicking::panic_fmt::h74f9fec01e0efc2d
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/core/src/panicking.rs:100:14
  18:     0x7fe5f5fd758c - core::panicking::panic_display::hf32e97c4b8dcc6f0
  19:     0x7fe5f76fb056 - rustc_errors::HandlerInner::flush_delayed::h43a5f157aed43c3a
  20:     0x7fe5f76f9941 - <rustc_errors::HandlerInner as core::ops::drop::Drop>::drop::hd4e01f0a02740eef
  21:     0x7fe5f6dbc8e6 - core::ptr::drop_in_place<rustc_session::parse::ParseSess>::h53e6b368104faf10
thread 'rustc' panicked at 'no warnings or errors encountered even though `delayed_good_path_bugs` issued', compiler/rustc_errors/src/lib.rs:1165:13
stack backtrace:
  22:     0x7fe5f6dbe6da - <alloc::rc::Rc<T> as core::ops::drop::Drop>::drop::h6f1b5ef2f5210a61
  23:     0x7fe5f6d99c7d - core::ptr::drop_in_place<rustc_interface::interface::Compiler>::h3abd6aaa3f9cc9d2
   0:     0x7f5d5e910afc - std::backtrace_rs::backtrace::libunwind::trace::h25fd93f23613b0bd
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x7f5d5e910afc - std::backtrace_rs::backtrace::trace_unsynchronized::h5c84b72802e6f97b
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f5d5e910afc - std::sys_common::backtrace::_print_fmt::h2963c3d3cac93165
  24:     0x7fe5f6d99a97 - rustc_span::with_source_map::h72b982789c82ce08
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f5d5e910afc - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hc11090b821447af0
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x7f5d5e96dd7c - core::fmt::write::h4177df205e3207df
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/core/src/fmt/mod.rs:1163:17
  25:     0x7fe5f6da8620 - scoped_tls::ScopedKey<T>::set::hc8b69f6cb27c0b84
  26:     0x7fe5f6d9a1f5 - std::sys_common::backtrace::__rust_begin_short_backtrace::h0b7220eff104d5d7
  27:     0x7fe5f6dbc342 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hc98bdb0dec4259f0
   5:     0x7f5d5e901065 - std::io::Write::write_fmt::h43248dd5a934053b
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/io/mod.rs:1696:15
   6:     0x7f5d5e913d50 - std::sys_common::backtrace::_print::he9298a8f9ec8c87d
  28:     0x7fe5f46ea853 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::heed06d1e1a2f1ff5
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:49:5
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/alloc/src/boxed.rs:1691:9
  29:     0x7fe5f46ea853 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h471e2149edf2c242
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/alloc/src/boxed.rs:1691:9
   7:     0x7f5d5e913d50 - std::sys_common::backtrace::print::h21f708b17f3753fb
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x7f5d5e913d50 - std::panicking::default_hook::{{closure}}::hbf61d78ce94d8717
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:210:50
  30:     0x7fe5f46ea853 - std::sys::unix::thread::Thread::new::thread_start::hf2c59eb01537ff66
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys/unix/thread.rs:106:17
  31:     0x7fe5f4478927 - start_thread
                               at ./nptl/./nptl/pthread_create.c:435:8
   9:     0x7f5d5e9138fb - std::panicking::default_hook::h17de5f695e938d3c
  32:     0x7fe5f45089e4 - __clone
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:227:9
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:100
  33:                0x0 - <unknown>

  10:     0x7f5d5f0f45c1 - rustc_driver::DEFAULT_HOOK::{{closure}}::{{closure}}::h136d34559455780f
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.57.0-beta.2 (7e4c9eebd 2021-10-22) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z symbol-mangling-version=v0 -Z macro-backtrace -Z tls-model=initial-exec -Z unstable-options -Z binary-dep-depinfo -Z force-unstable-if-unmarked -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 -C linker=/usr/bin/clang -C link-args=-Wl,-rpath,$ORIGIN/../lib -C prefer-dynamic --crate-type lib

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
thread panicked while panicking. aborting.
  11:     0x7f5d52dd9819 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h85db76da50bfa969
  12:     0x7f5d52dda76b - proc_macro::bridge::client::<impl proc_macro::bridge::Bridge>::enter::{{closure}}::{{closure}}::h1b552d664e4eed54
  13:     0x7f5d5e914569 - std::panicking::rust_panic_with_hook::hab1577e24779214f
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:628:17
  14:     0x7f5d5e914020 - std::panicking::begin_panic_handler::{{closure}}::h8c53778a9a259922
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:521:13
  15:     0x7f5d5e910fa4 - std::sys_common::backtrace::__rust_end_short_backtrace::h24cd09d06b6955fd
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys_common/backtrace.rs:139:18
  16:     0x7f5d5e913f89 - rust_begin_unwind
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/panicking.rs:517:5
  17:     0x7f5d5e8d9261 - core::panicking::panic_fmt::h74f9fec01e0efc2d
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/core/src/panicking.rs:100:14
  18:     0x7f5d6020c58c - core::panicking::panic_display::hf32e97c4b8dcc6f0
  19:     0x7f5d61930056 - rustc_errors::HandlerInner::flush_delayed::h43a5f157aed43c3a
  20:     0x7f5d6192e941 - <rustc_errors::HandlerInner as core::ops::drop::Drop>::drop::hd4e01f0a02740eef
  21:     0x7f5d60ff18e6 - core::ptr::drop_in_place<rustc_session::parse::ParseSess>::h53e6b368104faf10
  22:     0x7f5d60ff36da - <alloc::rc::Rc<T> as core::ops::drop::Drop>::drop::h6f1b5ef2f5210a61
  23:     0x7f5d60fcec7d - core::ptr::drop_in_place<rustc_interface::interface::Compiler>::h3abd6aaa3f9cc9d2
  24:     0x7f5d60fcea97 - rustc_span::with_source_map::h72b982789c82ce08
  25:     0x7f5d60fdd620 - scoped_tls::ScopedKey<T>::set::hc8b69f6cb27c0b84
  26:     0x7f5d60fcf1f5 - std::sys_common::backtrace::__rust_begin_short_backtrace::h0b7220eff104d5d7
  27:     0x7f5d60ff1342 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hc98bdb0dec4259f0
  28:     0x7f5d5e91f853 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::heed06d1e1a2f1ff5
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/alloc/src/boxed.rs:1691:9
  29:     0x7f5d5e91f853 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h471e2149edf2c242
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/alloc/src/boxed.rs:1691:9
  30:     0x7f5d5e91f853 - std::sys::unix::thread::Thread::new::thread_start::hf2c59eb01537ff66
                               at /rustc/7e4c9eebd82e9fa71f74626e5ba4e3494b8aba25/library/std/src/sys/unix/thread.rs:106:17
  31:     0x7f5d5e6ad927 - start_thread
                               at ./nptl/./nptl/pthread_create.c:435:8
  32:     0x7f5d5e73d9e4 - __clone
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:100
  33:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.57.0-beta.2 (7e4c9eebd 2021-10-22) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z symbol-mangling-version=v0 -Z macro-backtrace -Z tls-model=initial-exec -Z unstable-options -Z binary-dep-depinfo -Z force-unstable-if-unmarked -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 -C linker=/usr/bin/clang -C link-args=-Wl,-rpath,$ORIGIN/../lib -C prefer-dynamic

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
thread panicked while panicking. aborting.
rustc exited with signal: 4 (core dumped)
error: could not compile `rustc_mir_build`

Caused by:
  process didn't exit successfully: `/home/user/src/rust/build/bootstrap/debug/rustc --crate-name rustc_mir_build --edition=2021 compiler/rustc_mir_build/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata -C opt-level=3 -C embed-bitcode=no -C debuginfo=0 -C metadata=ee96a5960d5e7a6a -C extra-filename=-ee96a5960d5e7a6a --out-dir /home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -C linker=/usr/bin/clang -L dependency=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps -L dependency=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/release/deps --extern rustc_apfloat=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_apfloat-fddfbe1731e6cd56.rmeta --extern rustc_arena=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_arena-2cea5c898bd94dbb.rmeta --extern rustc_ast=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_ast-ea7d1ac19b521674.rmeta --extern rustc_attr=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_attr-e3ccebe0f95b1bc7.rmeta --extern rustc_data_structures=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_data_structures-4b67691203e46120.rmeta --extern rustc_errors=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_errors-fd7bbe56e027f43c.rmeta --extern rustc_hir=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_hir-299319cc08bb44b5.rmeta --extern rustc_index=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_index-f733038efaf72913.rmeta --extern rustc_infer=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_infer-25a25ae4146f7d7a.rmeta --extern rustc_middle=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_middle-ef1a2363ef493d6b.rmeta --extern rustc_serialize=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_serialize-1a43f7785dc38fe4.rmeta --extern rustc_session=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_session-e0fbf934abce7b76.rmeta --extern rustc_span=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_span-a2fa314d1db2ce2b.rmeta --extern rustc_target=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_target-cffa0ed18d4afbf9.rmeta --extern rustc_trait_selection=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_trait_selection-c2f0721a2a3d01dd.rmeta --extern smallvec=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libsmallvec-b6a50ba40fd51145.rmeta --extern tracing=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libtracing-5bd6b61ef83744a2.rmeta --cfg=bootstrap -Zsymbol-mangling-version=v0 -Zmacro-backtrace '-Clink-args=-Wl,-rpath,$ORIGIN/../lib' -Ztls-model=initial-exec -Zunstable-options '-Wrustc::internal' -Cprefer-dynamic -Z binary-dep-depinfo -L native=/home/user/src/rust/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/build/psm-b85bcfe9d1b8d998/out` (exit status: 254)
warning: build failed, waiting for other jobs to finish...
rustc exited with signal: 4 (core dumped)
error: build failed
Build completed unsuccessfully in 0:00:02
```

</p>
</details>

cc tracking issue  #87335
cc @cormacrelf because I'm not sure if #89841 fixes this ICE or not.

**Comment by est31** [comment]
I would like to minimize it further, but before that I'd like some feedback whether the ICE is fixed by #89841 or not.

**Comment by est31** [comment]
wait this is a dupe of #89960, the fix just didn't have time to get to the bootstrap compiler yet. Closing.
